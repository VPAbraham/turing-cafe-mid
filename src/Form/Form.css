.form {
  display: flex;
}

input {
  margin: 0 20px;
  width: 200px;
  height: 20px;
}

button {
  width: 200px;
  background: #008080;
  color: white;
}