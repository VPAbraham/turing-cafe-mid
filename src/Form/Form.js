import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state ={
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value})
  }

  submitReservation = (event) => {
    event.preventDefault();
    const newRes = {
      ...this.state
    }
    this.props.addReservation(newRes); 
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({ name: '', date: '', time: '', number: '' });
  }

  render() {
    return (
      <form className="form">
        <input 
        type="text"
        placeholder="Name"
        name="name"
        value={this.state.name}
        onChange={event=>this.handleChange(event)}
        />
        <input
          type="text"
          placeholder="Date (mm/dd)"
          name="date"
          value={this.state.date}
          onChange={event => this.handleChange(event)}
        />
        <input
          type="text"
          placeholder="Time"
          name="time"
          onChange={event => this.handleChange(event)}
        />
        <input
          type="number"
          placeholder="Number of guests"
          name="number"
          value={this.state.number}
          onChange={event => this.handleChange(event)}
        />
        <button
        onClick={event => this.submitReservation(event)}
        >Make Reservation</button>
      </form>
    )
  }
}

export default Form;