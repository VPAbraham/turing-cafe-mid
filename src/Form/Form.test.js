import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper
  const addReservation = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<Form addReservation={addReservation} />
    )
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
    
  it('should set inputted text and date to state', () => {

  })

  it('should run addReservation on button click', () => {

  })
})
