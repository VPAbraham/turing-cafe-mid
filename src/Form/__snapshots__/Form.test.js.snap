// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
